index.d.ts

```diff
@@ -0,0 +1,745 @@
+/**
+ * An interface that describes what structure is expected from Snapdragon's Node.
+ */
+declare interface NodeLike<T> {
+    /**
+     * Property that is always present and true in Node.
+     */
+    readonly isNode: boolean;
+
+    /**
+     * [Optional] Value property.
+     */
+    //TODO: Which one is default or legacy?
+    value?: string;
+
+    /**
+     * Value property.
+     */
+    //TODO: Which one is default or legacy?
+    val: string;
+
+    /**
+     * [Optional] Node's children.
+     */
+    nodes?: NodeLike<T>[];
+
+    /**
+     * [Optional] Parent node.
+     */
+    parent?: NodeLike<T>;
+
+    /**
+     * Type property.
+     */
+    type: string;
+
+    /**
+     * [Optional] Unshift node to children node array (`this.nodes`).
+     */
+    unshift?: (node: NodeLike<T>) => number;
+
+    /**
+     * [Optional] Unshift node to children node array (`this.nodes`).
+     */
+    //TODO: Is this a legacy? Should it be removed?
+    unshiftNode?: (node: NodeLike<T>) => number;
+
```

Comment:
snapdragon-utils has an `.unshiftNode` method, but nodes themselves (instances of `snapdragon-node`, at least newer versions) have an `.unshift` method. When the method is actually on an instance of node, we drop the `Node` part of the method name, since it's already implicit. 

Same naming conventions for the other methods.

Reply:
`utils.addOpen` has a `node.unshiftNode` fallback method in it. That's why I thought that it was a legacy method. That's what got me confused, could it be a bug?